astropy/io/fits/hdu/compressed.py

```diff
@@ -1047,42 +1066,42 @@ def _update_header_data(
 
         naxis = self._image_header["NAXIS"]
 
-        if not tile_size:
-            tile_size = []
-        elif len(tile_size) != naxis:
+        if not tile_shape:
+            tile_shape = []
+        elif len(tile_shape) != naxis:
             warnings.warn(
                 "Provided tile size not appropriate for the data.  "
                 "Default tile size will be used.",
                 AstropyUserWarning,
             )
-            tile_size = []
+            tile_shape = []
 
         # Set default tile dimensions for HCOMPRESS_1
 
         if compression_type == "HCOMPRESS_1":
             if self._image_header["NAXIS1"] < 4 or self._image_header["NAXIS2"] < 4:
                 raise ValueError("Hcompress minimum image dimension is 4 pixels")
-            elif tile_size:
-                if tile_size[0] < 4 or tile_size[1] < 4:
+            elif tile_shape:
+                if tile_shape[-1] < 4 or tile_shape[-2] < 4:
```

Comment:
HCOMPRESS_1 is for 2D images so I would keep 0 and 1 for clarity, as this is equivalent to the previous code ? (and same thing below)

Reply:
It's actually possible (and supported in astropy.io.fits) to compress 3D+ data with HCOMPRESS as long as the tile size is 1 for all extra dimensions so unfortunately I think I have to leave it as I have it now?

Reply:
Ah OK, interesting, all good then.